source/shell-navigation-os.rst

```diff
@@ -18,295 +18,619 @@ Lesson 3: Shell Navigation
+  # declare STRING variable
+  STRING="Hello World"
+  # print variable on a screen
+  echo $STRING
 
-::
+Now make the script executable using ``chmod`` and run the script. What does it do?
 
-    $ ls b    # <tab>
-    $ ls ba   # <tab>
-    bar_thing/ baz_thing/
-    $ ls bar  # <tab>
-    $ ls bar_thing
+.. rst-class:: build
 
+.. code-block:: console
 
-TODO
-----
+  $ chmod +x hello_world.sh
+  $ ./hello_world.sh
+  Hello World
 
-.. TODO: Add activity
-.. TODO: Add Answer Key
-.. They've already changed their passwords, what else would be a good
-.. challenge?
+Passing arguments to the bash script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+When you pass arguments to a bash script, you can reference them inside of the script using ``$1``, ``$2``, etc. This
+means if you do something like '``foo.sh bar``', ``$1`` will return ``bar``. You can also use ``$@`` to reference all
+arguments.
+
+For example:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  echo $1   # prints argument #1 given to script
+  echo $@   # prints all arguments given to script
+
+Reading User Input
+~~~~~~~~~~~~~~~~~~
+
+You can also take input using the ``read`` command which then stores it in a variable. If you pass ``read`` '``-a``',
+it puts the input into a bash array.
+
+For example:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  echo -e "Tell me a word: \c"
+  read word
+  echo "Your word is $word"
+
+.. rst-class:: build
+
+.. code-block:: console
+
+  $ ./read.sh
+  Tell me a word: foo
+  Your word is foo
+
+Simple Bash if/else statement
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Bash conditionals use ``if``, ``else``, ``then`` and ``fi`` operators. You can compare strings, files and even command
+output. An example:
+
+.. code-block:: bash
+
+  #!/bin/bash
+  if [ "$1" == "foo" ] ; then
+    echo "You said $1"
+  else
+    echo "You did not say foo"
+  fi
+
+.. rst-class:: build
+
+.. code-block:: console
+
+  $ ./sayfoo.sh foo
+  You said foo
+  $ ./sayfoo.sh bar
+  You did not say foo
+
+Exercises: Bash
+---------------
+
+- Using the ``tar`` command, write a script named ``backup.sh`` which backs up the ``dobc`` user home directory into a
+  file ``/tmp/dobc-backup.tar.gz`` . *Hint: use the man page for tar or type 'tar -h'.*
+- Create a script called ``args.sh`` that takes three arguments and prints all of the args and then prints them in
+  reverse using ``echo``.
+- Create a script named ``input.sh`` which takes input for three args and then prints them in a sentence (of your
+  choosing).
+- Create a script named ``ifelse.sh`` which takes two arguments. If both arguments match, print ``"Yay, they match!"``,
+  if they don't, then print ``"Boo, they don't match :("``.
+
+.. ifslides::
+
+	*We likely won't have enough time to cover these all, but feel free to try later!*
+
+Exercise Answer Key
+-------------------
+
+Simple Backup script
+~~~~~~~~~~~~~~~~~~~~
+
+.. rst-class:: build
+
+.. code-block:: bash
+
+  #!/bin/bash
+  tar -vczf /tmp/dobc-backup.tar.gz /home/dobc
```

Comment:
It might be nice to explain what each of these flags mean:

v - verbose
c - compress
z - use gzip
f - output to file